quickstart/daml/licensing/daml/Licensing/License.daml

```diff
@@ -1,148 +1,189 @@
-               amuletReceiverAmounts = [ReceiverAmuletAmount provider licenseFeeCc]
-               dso 
-               lockedAmulet = acceptedPayment.lockedAmulet
-               round = openMiningRound.round
-               reference
-         acceptedPayment === expectedAcceptedPayment
-         license <- fetchAndArchive Group {user, provider, dso} licenseCid
-         licenseNum === license.licenseNum
-         now <- getTime
-         create License with
-                    dso
-                    user
-                    provider
-                    licenseNum
-                    params = license.params
-                    expiresAt = (max now license.expiresAt) `addRelTime` licenseExtensionDuration       
-
--- | A license contract between an app provider and an app user to use the app
--- for a specific purpose.
-template License with
-    provider : Party -- ^ App provider issuing the license.
-    user : Party -- ^ App user holding the license.
-    dso : Party -- ^ DSO party issuing the CC to be used for billing.
-    expiresAt: Time -- ^ Time when the license expires.
-    licenseNum : Int -- ^ License number identifying this license among all of the ones granted by the provider to this user.
-    params : LicenseParams -- ^ Parameters determining the behavior and purpose of the license.
-  where
-    signatory provider, user
-    
-    nonconsuming choice License_Renew : (ContractId LicenseRenewalRequest, ContractId AppPaymentRequest)
-      with
-        licenseFeeCc: Decimal      
-        licenseExtensionDuration : RelTime
-        paymentAcceptanceDuration: RelTime
-        description: Text
-      controller provider
-      do now <- getTime
-         paymentRequest <- create AppPaymentRequest with
-           provider, dso
-           sender = user
-           receiverAmounts = [ReceiverAmount provider (PaymentAmount licenseFeeCc AmuletUnit)]
-           description
-           expiresAt = now `addRelTime` paymentAcceptanceDuration
-         renewalOffer <- create LicenseRenewalRequest
-           with
-             provider, dso, user 
-             licenseFeeCc
-             licenseExtensionDuration
-             licenseNum
-             reference = paymentRequest
-         pure (renewalOffer, paymentRequest)    
+        assertWithinDeadline "settleBefore" settleBefore
 
+        alloc <- fetch @Allocation allocationCid
+        let allocView = view @Allocation alloc
+            transferLeg = allocView.allocation.transferLeg
+            settlementRef = (view $ toInterface @AllocationRequest this).settlement.settlementRef
+
+        allocView.allocation.settlement.settlementRef === settlementRef
+        transferLeg.instrumentId === licenseFeeInstrumentId
+        transferLeg.amount === licenseFeeAmount
+        transferLeg.receiver === provider
+        transferLeg.sender === user
+
+        exercise allocationCid (Allocation_ExecuteTransfer extraArgs)
+
+        oldLicense <- fetch licenseCid
```

Comment:
```suggestion
        oldLicense <- fetch licenseCid
        archive licenseCid
```

style nit: forgetting to archive a contract is quite easy. This way its clear that it will be archived.